### ISSUE TITLE
Versioning strategy for 0.x

### ISSUE THREAD

**Issue by cmeeren** [Zaid-Ajaj/Feliz#71]
As mentioned in https://github.com/cmeeren/Feliz.MaterialUI/issues/23#issuecomment-539325618 I lock the Feliz minor version when releasing Feliz.MaterialUI, under the assumption that minor version bumps may contain breaking changes, while patch version bumps do not. The reason for locking the version is to avoid problems as described in that issue.

I was just wondering if it would be possible for you to adopt this strategy more or less consistently while on 0.x. In other words, bump the minor version only if there are breaking changes, and otherwise just bump the patch version (if all changes are backwards-compatible). Feliz releases so often with minor version bumps that I don't have the capacity to track this and re-release Feliz.MaterialUI each time.

My suggestion is predicated on my (possibly incorrect) guess that many of Feliz's minor version bumps only contain backwards compatible changes (hard to know for sure, since I can't find any changelog/release notes). If that isn't the case, this whole point is moot.

**Comment by Shmew** [comment]
Is it possible to do something like a nightly build or hook triggered recompilation? If your library doesn't depend too strongly on Feliz that may be the easiest way to save time in the long run. That would potentially depend on having a good way to ensure things work in your build pipeline. I'm not sure how easy/viable an option that is.

**Comment by cmeeren** [comment]
Would that help? Won't the problems described in https://github.com/cmeeren/Feliz.MaterialUI/issues/23 only occur if there is a breaking change in the APIs used by Feliz.MaterialUI, which requires manual intervention anyway? (I may be wrong.)

In any case, I have no idea how to set that up. Sounds complex; it would (ideally) have to bump version, update release notes, commit changes back to the repo, etc.

The versioning strategy I described sounds like the simplest option, as it allows Feliz.MaterialUI to be forward-compatible with new versions of Feliz as long as the versioning scheme for breaking changes is known.

**Comment by Shmew** [comment]
Yeah, I agree it is the simplest.

What I was thinking of was that you could have a CI triggered build re-compile and try to compile a demo app and if it didn't fail it could be pushed out automatically. It would let you continue to pin against whatever version you compile against. What it does it lets you know when it breaks so you can fix the issue if it's on your end. In the case of the issue I had logged it was just MaterialUI needed to be regenerated/packaged for the new version.

 The build scripts I used in my repo do all of that except I update the release notes manually, might not be too much work with that to start with.

**Comment by Zaid-Ajaj** [comment]
> I was just wondering if it would be possible for you to adopt this strategy more or less consistently while on 0.x. In other words, bump the minor version only if there are breaking changes, and otherwise just bump the patch version (if all changes are backwards-compatible). 

Usually this is how I do it: 
 - Increment patch when fixing bugs in internal implementation
 - Increment minor when adding functionality to the public surface API that are backwards-compatible 
- Increment major when there are breaking changes to the surface API

Though here since I am still at 0.x I am incrementing minor loosely and inconsistently for whatever change I make (since there is no major release yet): sorry for this, I will try to adopt your strategy more consistently and also introduce a change log.

In summary: increment minor for breaking changes, increment patch for bug fixes and small changes such as docs or the addition of a new function overload

**Comment by cmeeren** [comment]
> Usually this is how I do it:
> 
>     * Increment patch when fixing bugs in internal implementation
> 
>     * Increment minor when adding functionality to the public surface API that are backwards-compatible
> 
>     * Increment major when there are breaking changes to the surface API

Yes, this is definitely the way to do it after 1.x

> In summary: increment minor for breaking changes, increment patch for bug fixes and small changes such as docs or the addition of a new function overload

Thanks a lot! 👍
